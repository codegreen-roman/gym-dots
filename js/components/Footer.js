import React from 'react'

const style = {
    border: '1px solid #FF00FF',
    margin: '2rem'
}

export const Footer = () => {
    return (
        <section style={style}>Hi, I am component Footer</section>
    )
}
