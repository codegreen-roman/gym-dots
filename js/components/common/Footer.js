import React from 'react'
import { compose, merge } from 'ramda'
import { connect } from 'react-redux'
import { shape, string, func } from 'prop-types'
import { loadLocationForUsername } from '../../state/actions/creators'

const style = {
    border: '1px solid #FF00FF',
    margin: '2rem'
}

const blockStyle = status => {

    const mergeStyle = merge(style)

    return mergeStyle({
        opacity: status === 'ready' ? '0.9' : '0.1'
    })
}

const _Footer = ({ githubLocation, fireAction }) => {

    const handleClick = () => fireAction(githubLocation.username)

    return (
        <section style={blockStyle(githubLocation.status)}>
            Hi, I am component Footer
            <button onClick={handleClick}>Request</button>
            <div>{githubLocation.username}</div>
            <div>{githubLocation.location}</div>
        </section>
    )
}

_Footer.propTypes = {
    githubLocation: shape({
        username: string.isRequired,
        location: string.isRequired,
        status: string
    }),
    fireAction: func.isRequired
}

const mapStateToProps = ({ githubLocation }) => ({
    githubLocation
})

const mapActionsToProps = dispatch => {
    return {
        fireAction: compose(dispatch, loadLocationForUsername)
    }
}

export { _Footer }
export const Footer = connect(mapStateToProps, mapActionsToProps)(_Footer)
