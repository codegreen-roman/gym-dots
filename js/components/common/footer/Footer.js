import React from 'react'
import { connect } from 'react-redux'
import { FooterButton } from './buttons/FooterButton'
import { footerStyle } from './Footer.glamor'
import { startWorkoutWithCountdown, loadAppDefaults, subscribeToAppDefaultsChanges } from '../../../state/actions/index'
import { compose } from 'ramda'
import { func, bool } from 'prop-types'
import { Flex } from 'glamorous-jsxstyle'

export class _Footer extends React.Component {

    componentDidMount() {
        const { loadDefaults } = this.props
        loadDefaults()
    }

    renderTrainingButtons() {

        const { fireStartWorkout, blocked, training } = this.props
        const buttonTitle = blocked ? 'Starting ...' : 'Start Workout'

        if (training) {
            return (
                <Flex>
                    <FooterButton data-test='failButton' disabled={false}>Failed</FooterButton>
                    <FooterButton data-test='doneButton' disabled={false}>Done</FooterButton>
                </Flex>
            )
        }

        return (
            <Flex>
                <FooterButton disabled={blocked} clickHandler={fireStartWorkout}>{buttonTitle}</FooterButton>
            </Flex>
        )
    }

    render() {

        const { hidden } = this.props

        if (hidden) return null

        return (
            <footer {...footerStyle}>
                {this.renderTrainingButtons()}
            </footer>
        )
    }
}

_Footer.propTypes = {
    blocked: bool.isRequired,
    training: bool.isRequired,
    hidden: bool.isRequired,
    fireStartWorkout: func.isRequired,
    loadDefaults: func.isRequired
}

const mapStateToProps = ({ workoutStatus }) => ({
    hidden: false,
    blocked: workoutStatus === 'starting',
    training: workoutStatus === 'started',
})

const mapActionsToProps = (dispatch, { history }) => {

    const startWorkout = compose(dispatch, startWorkoutWithCountdown)

    subscribeToAppDefaultsChanges(dispatch)

    return {
        fireStartWorkout: () => {
            startWorkout()
                .then(() => history.push('/activity/workout'))
        },
        loadDefaults: compose(dispatch, loadAppDefaults)
    }
}

export const Footer = connect(mapStateToProps, mapActionsToProps)(_Footer)
