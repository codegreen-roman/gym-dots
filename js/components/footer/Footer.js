import React from 'react'
import { connect } from 'react-redux'
import glamorous from 'glamorous'
import { footerStyle } from './Footer.glamor'
import { startWorkoutWithCountdown, loadAppDefaults, subscribeToAppDefaultsChanges } from '../../state/actions'
import { compose } from 'ramda'
import { func, bool, object } from 'prop-types'

const AButton = glamorous.button({
    flex: 1
})

export class _Footer extends React.Component {

    componentDidMount() {
        const { loadDefaults } = this.props
        loadDefaults()
    }

    render() {

        const { fireStartWorkout, blocked, history } = this.props
        const handleClick = () => {
            fireStartWorkout()
                .then(() => history.push('/workout/started'))
        }

        const buttonTitle = blocked ? 'Starting ...' : 'Start Workout'

        return (
            <footer {...footerStyle}>
                <AButton disabled={blocked} onClick={handleClick}>{buttonTitle}</AButton>
            </footer>
        )
    }
}

_Footer.propTypes = {
    history: object,
    blocked: bool.isRequired,
    fireStartWorkout: func.isRequired,
    loadDefaults: func.isRequired
}

const mapStateToProps = ({ workoutStatus }) => ({
    blocked: workoutStatus === 'starting'
})

const mapActionsToProps = dispatch => {

    subscribeToAppDefaultsChanges(dispatch)

    return {
        fireStartWorkout: compose(dispatch, startWorkoutWithCountdown),
        loadDefaults: compose(dispatch, loadAppDefaults)
    }
}

export const Footer = connect(mapStateToProps, mapActionsToProps)(_Footer)
