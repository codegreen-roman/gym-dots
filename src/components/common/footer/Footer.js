import React from 'react'
import { FooterWorkoutStart, FooterWorkoutControls } from './FooterControls'
import { T, always, cond } from 'ramda'
import { footerStyles } from './Footer.glamor'
import glamorous from 'glamorous'

/* Conditions */
export const isTrainingInAction = (props) => props.training
export const isTrainingAboutToStart = (props) => !props.training && !props.hidden
export const isFooterHidden = (props) => props.hidden

/* Conditional render */
export const conditionalRender = (props) => cond([
    [isTrainingInAction, always(<FooterWorkoutControls key='control-btns' {...props} />)],
    [isTrainingAboutToStart, always(<FooterWorkoutStart key='start-btn' {...props} />)],
    [isFooterHidden, always(null)],
    [T, always(null)]
])(props)

/* Footer */
export const Footer = (props) =>
    <StyledFooter>
        {conditionalRender(props)}
    </StyledFooter>

/* Footer styling logic */
const StyledFooter = glamorous.footer(
    footerStyles,
    ({children}) => children ? { padding: '15px'} : { padding: 0 }
)
