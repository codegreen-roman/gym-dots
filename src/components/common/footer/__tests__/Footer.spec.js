import React from 'react'
import { Footer } from '../Footer'
import { mount, render, shallow } from 'enzyme'

const setup = (blocked = false, training = false, shouldEndExercise = false) => {

    const props = {
        fireStartWorkout: jest.fn(),
        loadDefaults: jest.fn(),
        onSetFailed: jest.fn(),
        onSetDone: jest.fn(),
        fireCompleteExercise: jest.fn(),
        hidden: false,
        shouldEndExercise,
        blocked,
        training,
        currentResults: [true, true, true, true, true],
        nextExercise: {
            exerciseKey: 'thisIsTheUniqueKey'
        }
    }

    return {
        component: mount(<Footer {...props} />),
        wrapper: render(<Footer {...props} />),
        shallowWrapper: shallow(<Footer {...props} />),
        props
    }
}

describe('Footer component', () => {

    describe('snapshot not blocked', () => {

        it('matches the mounted snapshot', () => {
            const { component } = setup()
            expect(component).toMatchSnapshot()
        })

        it('matches the rendered snapshot', () => {
            const { wrapper } = setup()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe('snapshot blocked', () => {
        it('matches the previous snapshot', () => {
            const { component } = setup(true)
            expect(component).toMatchSnapshot()
        })
    })

    describe.skip('clicking on AButton', () => {
        it('should call fireStartWorkout function', () => {
            const { component, props } = setup()
            component.find('button').last().simulate('click')

            expect(props.fireStartWorkout).toHaveBeenCalledTimes(1)
        })
    })

    describe('in training status', () => {

        it('should show 2 buttons', () => {
            const { component } = setup(false, true)
            const buttons = component.find('button')

            expect(buttons.length).toBe(2)

        })

        it('show failed button', () => {
            const { wrapper } = setup(false, true)

            const button = wrapper.find('button').first()
            expect(button.text()).toBe('Failed')
        })

        it('show done button', () => {
            const { wrapper } = setup(false, true)

            const button = wrapper.find('button').last()
            expect(button.text()).toBe('Done')
        })

        it('matches the previous snapshot', () => {
            const { component } = setup(false, true)
            expect(component).toMatchSnapshot()
        })
    })

    describe.skip('shouldEndExercise prop is true', () => {
        it('should call fireCompleteExercise function', () => {

            const { props, shallowWrapper } = setup(false, true, false)

            shallowWrapper.setProps({ shouldEndExercise: true }, () => {
                expect(props.fireCompleteExercise).toHaveBeenCalledTimes(1)
                expect(props.fireCompleteExercise).toHaveBeenCalledWith('thisIsTheUniqueKey', [true, true, true, true, true])
            })

        })
    })

})
