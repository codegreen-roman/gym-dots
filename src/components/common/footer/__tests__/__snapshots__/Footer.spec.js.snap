// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component .conditionalRender should render FooterWorkoutControls 1`] = `
<FooterWorkoutControls
  blocked={false}
  currentResults={
    Array [
      true,
      true,
      true,
      true,
      true,
    ]
  }
  fireCompleteExercise={[Function]}
  fireStartWorkout={[Function]}
  hidden={false}
  loadDefaults={[Function]}
  nextExercise={
    Object {
      "exerciseKey": "thisIsTheUniqueKey",
    }
  }
  onSetDone={[Function]}
  onSetFailed={[Function]}
  shouldEndExercise={false}
  training={true}
/>
`;

exports[`Footer component .conditionalRender should render FooterWorkoutStart 1`] = `
<withHandlers(_FooterWorkoutStart)
  blocked={false}
  currentResults={
    Array [
      true,
      true,
      true,
      true,
      true,
    ]
  }
  fireCompleteExercise={[Function]}
  fireStartWorkout={[Function]}
  hidden={false}
  loadDefaults={[Function]}
  nextExercise={
    Object {
      "exerciseKey": "thisIsTheUniqueKey",
    }
  }
  onSetDone={[Function]}
  onSetFailed={[Function]}
  shouldEndExercise={false}
  training={false}
/>
`;

exports[`Footer component .conditionalRender should render null 1`] = `null`;

exports[`Footer component in training status matches the previous snapshot 1`] = `
.css-1eaczue,
[data-css-1eaczue] {
  position: absolute;
  bottom: 0px;
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  font-family: sans-serif;
  background: -webkit-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: -moz-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  padding: 15px;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
}

.css-8pmgdu,
[data-css-8pmgdu] {
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  font-size: 20px;
  line-height: 24px;
  font-weight: 500;
  text-align: center;
  color: #46B4E3;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  height: 40px;
  justify-content: center;
  align-items: stretch;
  border: none;
  background: none;
  overflow: hidden;
  text-decoration: none;
  transition: all 300ms;
  outline: none;
  padding: 0px;
  letter-spacing: 1px;
  background-color: transparent;
  margin: 0 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-transition: all 300ms;
  -moz-transition: all 300ms;
}

<Footer
  blocked={false}
  currentResults={
    Array [
      true,
      true,
      true,
      true,
      true,
    ]
  }
  fireCompleteExercise={[Function]}
  fireStartWorkout={[Function]}
  hidden={false}
  loadDefaults={[Function]}
  nextExercise={
    Object {
      "exerciseKey": "thisIsTheUniqueKey",
    }
  }
  onSetDone={[Function]}
  onSetFailed={[Function]}
  shouldEndExercise={false}
  training={true}
>
  <glamorous(footer)>
    <footer
      className="css-1eaczue"
    >
      <FooterWorkoutControls
        blocked={false}
        currentResults={
          Array [
            true,
            true,
            true,
            true,
            true,
          ]
        }
        fireCompleteExercise={[Function]}
        fireStartWorkout={[Function]}
        hidden={false}
        key="control-btns"
        loadDefaults={[Function]}
        nextExercise={
          Object {
            "exerciseKey": "thisIsTheUniqueKey",
          }
        }
        onSetDone={[Function]}
        onSetFailed={[Function]}
        shouldEndExercise={false}
        training={true}
      >
        <Aux>
          <glamorous(Button)
            data-test="failButton"
            disabled={false}
            onClickAction={[Function]}
          >
            <Button
              className="css-8pmgdu"
              data-test="failButton"
              disabled={false}
              onClickAction={[Function]}
            >
              <button
                className="css-8pmgdu"
                data-test="failButton"
                disabled={false}
                onClick={[Function]}
              >
                Failed
              </button>
            </Button>
          </glamorous(Button)>
          <glamorous(Button)
            data-test="doneButton"
            disabled={false}
            onClickAction={[Function]}
          >
            <Button
              className="css-8pmgdu"
              data-test="doneButton"
              disabled={false}
              onClickAction={[Function]}
            >
              <button
                className="css-8pmgdu"
                data-test="doneButton"
                disabled={false}
                onClick={[Function]}
              >
                Done
              </button>
            </Button>
          </glamorous(Button)>
        </Aux>
      </FooterWorkoutControls>
    </footer>
  </glamorous(footer)>
</Footer>
`;

exports[`Footer component snapshot blocked matches the previous snapshot 1`] = `
.css-1eaczue,
[data-css-1eaczue] {
  position: absolute;
  bottom: 0px;
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  font-family: sans-serif;
  background: -webkit-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: -moz-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  padding: 15px;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
}

.css-8pmgdu,
[data-css-8pmgdu] {
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  font-size: 20px;
  line-height: 24px;
  font-weight: 500;
  text-align: center;
  color: #46B4E3;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  height: 40px;
  justify-content: center;
  align-items: stretch;
  border: none;
  background: none;
  overflow: hidden;
  text-decoration: none;
  transition: all 300ms;
  outline: none;
  padding: 0px;
  letter-spacing: 1px;
  background-color: transparent;
  margin: 0 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-transition: all 300ms;
  -moz-transition: all 300ms;
}

<Footer
  blocked={true}
  currentResults={
    Array [
      true,
      true,
      true,
      true,
      true,
    ]
  }
  fireCompleteExercise={[Function]}
  fireStartWorkout={[Function]}
  hidden={false}
  loadDefaults={[Function]}
  nextExercise={
    Object {
      "exerciseKey": "thisIsTheUniqueKey",
    }
  }
  onSetDone={[Function]}
  onSetFailed={[Function]}
  shouldEndExercise={false}
  training={false}
>
  <glamorous(footer)>
    <footer
      className="css-1eaczue"
    >
      <withHandlers(_FooterWorkoutStart)
        blocked={true}
        currentResults={
          Array [
            true,
            true,
            true,
            true,
            true,
          ]
        }
        fireCompleteExercise={[Function]}
        fireStartWorkout={[Function]}
        hidden={false}
        key="start-btn"
        loadDefaults={[Function]}
        nextExercise={
          Object {
            "exerciseKey": "thisIsTheUniqueKey",
          }
        }
        onSetDone={[Function]}
        onSetFailed={[Function]}
        shouldEndExercise={false}
        training={false}
      >
        <_FooterWorkoutStart
          blocked={true}
          currentResults={
            Array [
              true,
              true,
              true,
              true,
              true,
            ]
          }
          fireCompleteExercise={[Function]}
          fireStartWorkout={[Function]}
          hidden={false}
          loadDefaults={[Function]}
          nextExercise={
            Object {
              "exerciseKey": "thisIsTheUniqueKey",
            }
          }
          onSetDone={[Function]}
          onSetFailed={[Function]}
          shouldEndExercise={false}
          startButtonHandler={[Function]}
          training={false}
        >
          <glamorous(Button)
            data-test="startButton"
            disabled={true}
            onClickAction={[Function]}
          >
            <Button
              className="css-8pmgdu"
              data-test="startButton"
              disabled={true}
              onClickAction={[Function]}
            >
              <button
                className="css-8pmgdu"
                data-test="startButton"
                disabled={true}
                onClick={[Function]}
              >
                Starting ...
              </button>
            </Button>
          </glamorous(Button)>
        </_FooterWorkoutStart>
      </withHandlers(_FooterWorkoutStart)>
    </footer>
  </glamorous(footer)>
</Footer>
`;

exports[`Footer component snapshot not blocked matches the mounted snapshot 1`] = `
.css-1eaczue,
[data-css-1eaczue] {
  position: absolute;
  bottom: 0px;
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  font-family: sans-serif;
  background: -webkit-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: -moz-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  padding: 15px;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
}

.css-8pmgdu,
[data-css-8pmgdu] {
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  font-size: 20px;
  line-height: 24px;
  font-weight: 500;
  text-align: center;
  color: #46B4E3;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  height: 40px;
  justify-content: center;
  align-items: stretch;
  border: none;
  background: none;
  overflow: hidden;
  text-decoration: none;
  transition: all 300ms;
  outline: none;
  padding: 0px;
  letter-spacing: 1px;
  background-color: transparent;
  margin: 0 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-transition: all 300ms;
  -moz-transition: all 300ms;
}

<Footer
  blocked={false}
  currentResults={
    Array [
      true,
      true,
      true,
      true,
      true,
    ]
  }
  fireCompleteExercise={[Function]}
  fireStartWorkout={[Function]}
  hidden={false}
  loadDefaults={[Function]}
  nextExercise={
    Object {
      "exerciseKey": "thisIsTheUniqueKey",
    }
  }
  onSetDone={[Function]}
  onSetFailed={[Function]}
  shouldEndExercise={false}
  training={false}
>
  <glamorous(footer)>
    <footer
      className="css-1eaczue"
    >
      <withHandlers(_FooterWorkoutStart)
        blocked={false}
        currentResults={
          Array [
            true,
            true,
            true,
            true,
            true,
          ]
        }
        fireCompleteExercise={[Function]}
        fireStartWorkout={[Function]}
        hidden={false}
        key="start-btn"
        loadDefaults={[Function]}
        nextExercise={
          Object {
            "exerciseKey": "thisIsTheUniqueKey",
          }
        }
        onSetDone={[Function]}
        onSetFailed={[Function]}
        shouldEndExercise={false}
        training={false}
      >
        <_FooterWorkoutStart
          blocked={false}
          currentResults={
            Array [
              true,
              true,
              true,
              true,
              true,
            ]
          }
          fireCompleteExercise={[Function]}
          fireStartWorkout={[Function]}
          hidden={false}
          loadDefaults={[Function]}
          nextExercise={
            Object {
              "exerciseKey": "thisIsTheUniqueKey",
            }
          }
          onSetDone={[Function]}
          onSetFailed={[Function]}
          shouldEndExercise={false}
          startButtonHandler={[Function]}
          training={false}
        >
          <glamorous(Button)
            data-test="startButton"
            disabled={false}
            onClickAction={[Function]}
          >
            <Button
              className="css-8pmgdu"
              data-test="startButton"
              disabled={false}
              onClickAction={[Function]}
            >
              <button
                className="css-8pmgdu"
                data-test="startButton"
                disabled={false}
                onClick={[Function]}
              >
                Start Workout
              </button>
            </Button>
          </glamorous(Button)>
        </_FooterWorkoutStart>
      </withHandlers(_FooterWorkoutStart)>
    </footer>
  </glamorous(footer)>
</Footer>
`;

exports[`Footer component snapshot not blocked matches the rendered snapshot 1`] = `
.css-1eaczue,
[data-css-1eaczue] {
  position: absolute;
  bottom: 0px;
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  font-family: sans-serif;
  background: -webkit-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: -moz-linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  background: linear-gradient(-180deg, #ECF7FE 0%, #FFFFFF 38%);
  padding: 15px;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
}

.css-8pmgdu,
[data-css-8pmgdu] {
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  font-size: 20px;
  line-height: 24px;
  font-weight: 500;
  text-align: center;
  color: #46B4E3;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  height: 40px;
  justify-content: center;
  align-items: stretch;
  border: none;
  background: none;
  overflow: hidden;
  text-decoration: none;
  transition: all 300ms;
  outline: none;
  padding: 0px;
  letter-spacing: 1px;
  background-color: transparent;
  margin: 0 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-transition: all 300ms;
  -moz-transition: all 300ms;
}

<footer
  class="css-1eaczue"
>
  <button
    class="css-8pmgdu"
    data-test="startButton"
  >
    Start Workout
  </button>
</footer>
`;
